doc/source/development/building_from_source.rst

```diff
@@ -224,6 +224,36 @@ All cached entries can be viewed using ``cmake -LAH`` from a build directory.
     `CMAKE_SKIP_INSTALL_RPATH <https://cmake.org/cmake/help/latest/variable/CMAKE_SKIP_INSTALL_RPATH.html>`__
     variable is not set.
 
+Resource files embedding
+++++++++++++++++++++++++
+
+Starting with GDAL 3.11, if a C23-compatible compiler is used, such as
+clang >= 19 or GCC >= 15, it is possible to embed resource files inside
+the GDAL library, without relying on resource files to be available on the file
+system (such resource files are located through an hard-coded
+path at build time in ``${CMAKE_INSTALL_PREFIX}/share/gdal``, or at run-time
+through the :config:`GDAL_DATA` configuration option).
+
+The following CMake options control that behavior:
+
+.. option:: EMBED_RESOURCE_FILES=ON/OFF
+
+    .. versionadded:: 3.11
+
+    Default is OFF for shared library builds (BUILD_SHARED_LIBS=ON), and ON
+    for static library builds (BUILD_SHARED_LIBS=OFF).
+    When ON, resource files needed by GDAL will be embedded into the GDAL library
+    and/or relevant plugins.
+
+.. option:: USE_ONLY_EMBEDDED_RESOURCE_FILES=ON/OFF
+
+    .. versionadded:: 3.11
+
+    Even if EMBED_RESOURCE_FILES=ON, by default GDAL will still try to locate
```

Comment:
There are currently two options, either fallback to the use the embedded resource, or refuse to use the filesystem. Is there a "user-extensible" use-case with resources where "use the embedded resource if present, but fallback to look in the filesystem" would make sense?

And language wise, my gut feeling is that "by default" should be moved to after "on the file system" in this sentence.

Reply:
> where "use the embedded resource if present, but fallback to look in the filesystem" would make sense?

I fail to see the use case. "Use the embedded resource if present": the presence of the embedded ressource is fully determined by EMBED_RESOURCE_FILES=ON/OFF.

> And language wise, my gut feeling is that "by default" should be moved to after "on the file system" in this sentence.

thanks, fixed